fv3core/stencils/updatedzc.py

```diff
@@ -4,194 +4,47 @@
-#            enddo
-#         enddo
-#         do j=js1, je2
-#            do i=is1, ie1
-#               yfx(i,j) = vt(i,j,km) + (vt(i,j,km)-vt(i,j,km-1))*bot_ratio
-#            enddo
-#         enddo
-
-
 @gtscript.function
 def p_weighted_average_bottom(vel, dp0):
     ratio = dp0[-1] / (dp0[-2] + dp0[-1])
-    # return (1. + ratio ) * vel[0, 0, -1] - ratio * vel[0, 0, -2]
     return vel[0, 0, -1] + (vel[0, 0, -1] - vel[0, 0, -2]) * ratio
 
 
-#      else     ! compute domain
-#         int_ratio = 1./(dp0(k-1)+dp0(k))
-#         do j=js1, je1
-#            do i=is1, ie2
-#               xfx(i,j) = (dp0(k)*ut(i,j,k-1)+dp0(k-1)*ut(i,j,k))*int_ratio
-#            enddo
-#         enddo
-#         do j=js1, je2
-#            do i=is1, ie1
-#               yfx(i,j) = (dp0(k)*vt(i,j,k-1)+dp0(k-1)*vt(i,j,k))*int_ratio
-#            enddo
-#         enddo
-#      endif
-
-
 @gtscript.function
 def p_weighted_average_domain(vel, dp0):
-    # ratio = dp0 / ( dp0[-1] + dp0 )
-    # return ratio * vel[0, 0, -1] + (1. - ratio) * vel
     int_ratio = 1.0 / (dp0[-1] + dp0)
     return (dp0 * vel[0, 0, -1] + dp0[-1] * vel) * int_ratio
 
 
-#      do j=js-ng, je+ng
-#         do i=is-ng, ie+ng
-#            gz2(i,j) = gz(i,j,k)
-#         enddo
-#      enddo
-
-#      if (grid_type < 3) call fill_4corners(gz2, 1, bd, npx, npy, sw_corner, se_corner, ne_corner, nw_corner) # noqa: E501
-
-#      do j=js1, je1
-#         do i=is1, ie2
-#            if( xfx(i,j) > 0.0 ) then
-#                fx(i,j) = gz2(i-1,j)
-#            else
-#                fx(i,j) = gz2(i  ,j)
-#            endif
-#            fx(i,j) = xfx(i,j)*fx(i,j)
-#         enddo
-#      enddo
-
-#      if (grid_type < 3) call fill_4corners(gz2, 2, bd, npx, npy, sw_corner, se_corner, ne_corner, nw_corner) # noqa: E501
-
-#      do j=js1,je2
-#         do i=is1,ie1
-#            if( yfx(i,j) > 0.0 ) then
-#                fy(i,j) = gz2(i,j-1)
-#            else
-#                fy(i,j) = gz2(i,j)
-#            endif
-#            fy(i,j) = yfx(i,j)*fy(i,j)
-#         enddo
-#      enddo
-
-
 @gtscript.function
 def xy_flux(gz_x, gz_y, xfx, yfx):
     fx = xfx * (gz_x[-1, 0, 0] if xfx > 0.0 else gz_x)
     fy = yfx * (gz_y[0, -1, 0] if yfx > 0.0 else gz_y)
     return fx, fy
 
 
-#      do j=js1, je1
-#         do i=is1,ie1
-#            gz(i,j,k) = (gz2(i,j)*area(i,j) +  fx(i,j)- fx(i+1,j)+ fy(i,j)- fy(i,j+1)) & # noqa: E501
-#                      / (         area(i,j) + xfx(i,j)-xfx(i+1,j)+yfx(i,j)-yfx(i,j+1)) # noqa: E501
-#         enddo
-#      enddo
-
-#   end do
-
-# ! Enforce monotonicity of height to prevent blowup
-#   do i=is1, ie1
-#      do j=js1, je1
-#         ws(i,j) = ( zs(i,j) - gz(i,j,km+1) ) * rdt
-#      enddo
-#   end do
-
-#   do k = km, 1, -1
-#      do j=js1, je1
-#         do i=is1, ie1
-#            gz(i,j,k) = max( gz(i,j,k), gz(i,j,k+1) + dz_min )
-#         enddo
-#      enddo
-#   enddo
+@gtstencil
+def set_zero_2d(out_field: FloatFieldIJ):
```

Comment:
This could potentially be a common stencil like `copy_stencil`, but that can wait until needed.

Reply:
that's a good point! I can add that if you'd like, though the next refactor of this stencil will not use that, it'll all be 1 stencil and this hacky hack won't be needed